idf_build_apps/manifest/if_parser.py

```diff
@@ -225,7 +229,6 @@ def get_value(self, target: str, config_name: str) -> Any:
 BOOL_EXPR = infixNotation(
     BOOL_STMT,
     [
-        (AND, 2, opAssoc.LEFT, BoolAnd),
-        (OR, 2, opAssoc.LEFT, BoolOr),
+        (MatchFirst((AND, OR)), 2, opAssoc.LEFT, BoolOrAnd),
```

Comment:
I made these changes because this way you can count how many `and` and `or` operators are in a single logic level. After doing this, I just added some if statements to display a warning.